### PR TITLE
Task 80130 extend validation to selection rule

### PR DESCRIPTION
**Pull request by mszymeshop**
### What
The validation for SelectionRule has been added now. It was missing before and validation focused on endpoints only.

### CODE REVIEW

File: src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using CaptainHook.Domain.Results;
 using CaptainHook.Domain.ValueObjects;
 using Eshopworld.Core;
+using Kusto.Cloud.Platform.Utils;
```
Comment (tabman83) [inline src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs:13]: Is it needed?

File: src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs [hunk 2]
```diff
@@ -34,12 +35,12 @@ public class UpsertWebhookRequestHandler : IRequestHandler<UpsertWebhookRequest,
...
             IDirectorServiceProxy directorService,
             IBigBrother bigBrother,
-            TimeSpan[] _sleepDurations = null)
+            TimeSpan[] sleepDurations = null)
         {
             _subscriberRepository = subscriberRepository;
             _directorService = directorService;
             _bigBrother = bigBrother;
-            _retrySleepDurations = _sleepDurations ?? DefaultRetrySleepDurations;
+            _retrySleepDurations = sleepDurations?.SafeFastNullIfEmpty() ?? DefaultRetrySleepDurations;
```
Comment (tabman83) [inline src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs:43]: SafeFastNullIfEmpty - What is that?

Comment (mszymeshop) [reply src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs:43]: it returns null if the collection is empty and by default, it is empty (by IOC) and I am using this extension method from Kusto here to address this.

File: src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs (hunk 1 above)
Comment (mszymeshop) [reply src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs:13]: See the comment below.

File: src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs (hunk 2 above)
Comment (chomick) [reply src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs:43]: Are we using entire package only for that extension? :)

**Pull request merged**